qiskit/transpiler/passes/scheduling/time_unit_conversion.py

```diff
@@ -135,3 +175,74 @@ def _unified(unit_set: Set[str]) -> str:
             return "SI"
 
         return "mixed"
+
+
+_DURATION_KIND_NAME = {
+    Duration.dt: "dt",
+    Duration.ns: "ns",
+    Duration.us: "us",
+    Duration.ms: "ms",
+    Duration.s: "s",
+}
+
+
+class _EvalDurationImpl(expr.ExprVisitor[float]):
+    """Evaluates the expression to a single float result.
+
+    If ``dt`` is provided or all durations are already in ``dt``, the result is in ``dt``.
+    Otherwise, the result will be in seconds, and all durations MUST be in wall-time (SI).
+    """
+
+    __slots__ = ("dt", "has_dt", "has_si")
+
+    def __init__(self, dt=None):
+        self.dt = dt if dt is not None else 1
+        self.has_dt = False
+        self.has_si = False
```

Comment:
Would leaving `self.dt = None` be unsafe here?  Feels like `1` is a technically valid value for `dt`.  Can't imagine that there are many QCs out there that have a cycle time of 1s, but still.

Reply:
Yeah, this was sort of gross for me to have done lol.

Another "divide by 1" for less code trick, which I'll change to just be local rather than at the instance level.